### ISSUE TITLE
Spell check not working

### ISSUE THREAD

**Issue by Koltsz** [IsmaelMartinez/teams-for-linux#677]
**Describe the bug**
Spell checking isn't working again

**To Reproduce**
N/A

**Expected behavior**
Spell check should be loaded and incorrect words should have a red underline

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Fedora 36 (Workstation Edition)
 - Installation package: Appimage and RPM
 - Version: 1.0.49

**Additional context**
N/A

**Comment by RolandRosenfeld** [comment]
Since I had similar problems in the past (with DEB package) on some of my computers, here's what helped for me:
I now hard coded the dictionary: 
`$ teams-for-linux --spellCheckerLanguages='de-DE'`
Since I do so, the correct dictionary is downloaded to ~/.config/teams-for-linux/Dictionaries/ and especially it is used (before  adding spellCheckerLanguages option, I had one computer always using the en-US dictionary instead of de-DE).

**Comment by Koltsz** [comment]
> Since I had similar problems in the past (with DEB package) on some of my computers, here's what helped for me: I now hard coded the dictionary: `$ teams-for-linux --spellCheckerLanguages='de-DE'` Since I do so, the correct dictionary is downloaded to ~/.config/teams-for-linux/Dictionaries/ and especially it is used (before adding spellCheckerLanguages option, I had one computer always using the en-US dictionary instead of de-DE).

Awesome, than you!
this fixed the issue

set mine to en-UK

**Comment by 22-all** [comment]
You can also set more than one language this way, for example `--spellCheckerLanguages='fr-CH' --spellCheckerLanguages='fr-FR' ` and it will happily check all languages.

**Comment by jijojosephk** [comment]
I would say there should be default values which is missing. This was broken when an option to set spellchecker language from command line was added.

**Comment by jijojosephk** [comment]
We'll have to set the default to `['en-US','en-UK']`

**Comment by jijojosephk** [comment]
Setting to `'en-US'`

**Comment by caio-hat** [comment]
For FlatPak Verctions the command order is different.
Edit the shortcut to Teams-For-Linux on your program panel
Add the flag **- -spellchecklanguages** at the end of the line

/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=teams-for-linux --file-forwarding com.github.IsmaelMartinez.teams_for_linux @@u %U @@ **--spellCheckerLanguages='pt-BR'**

**Comment by callegar** [comment]
Using flatpack, is it possible to pass two languages to `--spellCheckerLanguages`? It is unclear to me if I should pass some sort of *array* to the parameter...

**Comment by jijojosephk** [comment]
I think the developer's original intention was to have multiple languages. But it does not work, use only one language until we find a solution.

**Comment by callegar** [comment]
Actually using `--spellCheckerLanguages` twice seems to work. However passing two languages, words for both languages appear to be considered OK. Having the possibility to *switch* with the right click among either of the two languages or both would be great.

**Comment by jijojosephk** [comment]
> Actually using `--spellCheckerLanguages` twice seems to work. However passing two languages, words for both languages appear to be considered OK. Having the possibility to _switch_ with the right click among either of the two languages or both would be great.

I see, I had never tried. Your requirement might be possible. Right click on the text box or tray icon?

**Comment by jijojosephk** [comment]
Something like this?

![image](https://user-images.githubusercontent.com/11662148/232223419-b7a17253-adea-4312-b02b-61f5a6746574.png)

**Comment by saskra** [comment]
In case it helps anyone else, in the file `/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux.desktop/` I changed the line `Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux. desktop /snap/bin/teams-for-linux %U` to `Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux.desktop /snap/bin/teams-for-linux --spellCheckerLanguages='de-DE' %U`, which means that I now also have my desired dictionary available when starting Teams via Favorites (not via Terminal). However, I'm not very familiar with snap packages, so it's possible that it won't survive the next update, for example.

**Comment by callegar** [comment]
@jijojosephk
> Something like this?
Exactly! I understand it will be in the next release, thanks!!!

**Comment by jijojosephk** [comment]
> In case it helps anyone else, in the file `/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux.desktop/` I changed the line `Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux. desktop /snap/bin/teams-for-linux %U` to `Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/teams-for-linux_teams-for-linux.desktop /snap/bin/teams-for-linux --spellCheckerLanguages='de-DE' %U`, which means that I now also have my desired dictionary available when starting Teams via Favorites (not via Terminal). However, I'm not very familiar with snap packages, so it's possible that it won't survive the next update, for example.

It would be there.

**Comment by fumoretti** [comment]
> For flatpak version the command order is different. Edit the shortcut to Teams-For-Linux on your program panel Add the flag **- -spellchecklanguages** at the end of the line
> 
> /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=teams-for-linux --file-forwarding com.github.IsmaelMartinez.teams_for_linux @@U %U @@ **--spellCheckerLanguages='pt-BR'**

Thanks!! working well here with a little trick on KDE plasma. To avoid the shortcut replace during flatpak updates, i did a copy of .desktop file from "/var/lib/flatpak/..." to  "~/.local/share/applications/" and edit it including --spellCheckerLanguages=pt-BR at the end of Arguments line (yes, without ' ' ).

**Comment by callegar** [comment]
There is still one thing that is not completely clear to me. When the spellchecker identifies a word it does not know, it provides suggestions, but also an *add to dictionary* option. When you have multiple dictionaries active and the writing language is "all" how does the add to dictionary option work (i.e., what is the destination dictionary?).

**Comment by jijojosephk** [comment]
@ogarcia how do you setup spanish? using command line parameters right? The list in the menu is based on the parameters, so it should not break anything. Can you confirm?

**Comment by ogarcia** [comment]
@jijojosephk before this change it did nothing special, it just worked straight away and without any problems. After this change I have to launch the teams with the modifier `--spellCheckerLanguages='es-ES` (`$ teams-for-linux --spellCheckerLanguages='es-ES'`) for it to work correctly, if I don't do it, it doesn't allow me to deactivate the spell checker or to choose any other language than the two English ones.

**Comment by jijojosephk** [comment]
es-ES should be listed in the menu. Isn't it available?

**Comment by ogarcia** [comment]
> es-ES should be listed in the menu. Isn't it available?

No, it does not appear unless you start with the modifier `--spellCheckerLanguages='es-ES'` and in that case only `es-ES` appears as a possibility.

Start teams-for-linux without modifier:
![No es-ES](https://user-images.githubusercontent.com/572924/233632671-c89a2145-7a6c-453d-9daf-90bfa58b4a9a.png)

Start teams-for-linux with modifier:
![Only es-ES](https://user-images.githubusercontent.com/572924/233632795-56f04c57-f8ee-4429-a999-fe20e9a7cbb8.png)

**Comment by saskra** [comment]
>To avoid the shortcut replace during flatpak updates, i did a copy of .desktop file from "/var/lib/flatpak/..." to "~/.local/share/applications/" and edit it including --spellCheckerLanguages=pt-BR at the end of Arguments line

Thanks, I tried that too (albeit with the snap package), and it actually preserved the dictionary beyond the Teams update. But unfortunately the icon disappeared in the process because the path in the copied .desktop file was no longer correct. Somehow I also failed to get it back by putting in a fixed path to a copy of the icon.
